packages/liveblocks-react-blocknote/src/initialization/schema.ts

```diff
@@ -0,0 +1,53 @@
+import type {
+  BlockSchema,
+  InlineContentSchema,
+  StyleSchema,
+} from "@blocknote/core";
+import {
+  BlockNoteSchema,
+  createInlineContentSpecFromTipTapNode,
+} from "@blocknote/core";
+import { MentionNode } from "@liveblocks/react-tiptap";
+
+const mentionSpec = createInlineContentSpecFromTipTapNode(MentionNode, {
+  id: {
+    default: "",
+  },
+  notificationId: {
+    default: "",
+  },
+});
+/**
+ * Adds the Liveblocks Mention Node as inline content to the BlockNote schema
+ *
+ * This makes sure BlockNote knows about Liveblocks mentions and that you can read/write mentions via the BlockNote API
+ */
+export const withLiveblocksSchema = <
+  B extends BlockSchema,
+  I extends InlineContentSchema,
+  S extends StyleSchema,
+>(
+  schema?: BlockNoteSchema<B, I, S>,
+  liveblocksOptions: Partial<{ mentions: boolean }> = {}
+): BlockNoteSchema<B, I, S> => {
+  const optionalSchema = schema || BlockNoteSchema.create();
+  if (!liveblocksOptions.mentions) {
+    return optionalSchema as BlockNoteSchema<B, I, S>;
```

Comment:
@YousefED this was originally `as any`, in the parent file, I moved the type problem here. Our rules don't allow the use of `any`. I'm not sure if this type is correct. I guess we could have the return type be BlockNoteSchema<B,I,S> | DefaultBlocknoteSpec, but that's not exported from blocknote, it looks like it's constructed by `BlockNoteSchema<BlockSchemaFromSpecs<BSpecs>, InlineContentSchemaFromSpecs<ISpecs>, StyleSchemaFromSpecs<SSpecs>>` where BSpecs,ISpecs, and SSpecs are typeof default values.